joblib/parallel.py

```diff
@@ -681,11 +680,15 @@ def retrieve(self):
             with self._lock:
                 job = self._jobs.pop(0)
             try:
-                # check if timeout supported in backend future implementation
-                if 'timeout' in getfullargspec(job.get).args:
-                    self._output.extend(job.get(timeout=self.timeout))
+                if self.timeout is not None:
```

Comment:
+1 for doing backend introspection once and for all if the performance is good that way.